src/status_im2/contexts/chat/lightbox/utils.cljs

```diff
@@ -98,6 +99,20 @@
           (when (and enabled? (not= result orientation/landscape-right))
             (handle-orientation result props state animations))))))))
 
+(defn on-scroll
+  [e item-width images-opacity landscape?]
+  (let [total-item-width (+ item-width constants/separator-width)
+        progress         (/ (if landscape?
+                              (oops/oget e "nativeEvent.contentOffset.y")
+                              (oops/oget e "nativeEvent.contentOffset.x"))
+                            total-item-width)
+        index-initial    (max (Math/floor progress) 0)
+        index-final      (inc index-initial)
+        decimal-part     (- progress index-initial)]
+    (anim/set-val (nth images-opacity index-initial) (- 1 decimal-part))
+    (when (< index-final (count images-opacity))
+      (anim/set-val (nth images-opacity index-final) decimal-part))))
```

Comment:
Hello @OmarBasem 

Just a suggestion to change these lines to use `if` instead, IMO it is more clear, we have:

```clojure
(set-val (nth images idx) val)
(when some-condition
  (set-val (nth images idx) another-val))
```
 at the end of this function, the final value for `(nth images idx)` will depend on `some-condition`, so  we could use:
 
 ```clojure
 (if some-condition
  (set-val (nth images idx) another-val)
  (set-val (nth images idx) val))
 ```
 that I'd refactor including `decimal-part` as follows:
 ```clojure
 (let [,,,
      decimal-part (if some-condition
                    another-val
                    val)]
  (set-val (nth images-opacity index-final) decimal-part))
 ```
 
 I'm moving the condition to be solved in the `decimal-part` binding to highlight that the final, and single call of this function is to `anim/set-val`.
 
 wdyt?

Reply:
Hi @ulisesmac, I am not sure how this would work.

We need to animate two different values `index-initial` and `index-final` indices of the `images-opacity` list. We need to always animate both values, but there is a check for `index-final` for when we scroll to the last image and try to swipe even more it will try to read an index bigger than the list size.

Reply:
Yes! You are right!
I thought all animations were on a single value, not two.